cavern/src/main/java/org/opencadc/cavern/nodes/CephFSQuotaPlugin.java

```diff
@@ -0,0 +1,158 @@
+ *  the GNU Affero General Public        la “GNU Affero General Public
+ *  License as published by the          License” telle que publiée
+ *  Free Software Foundation,            par la Free Software Foundation
+ *  either version 3 of the              : soit la version 3 de cette
+ *  License, or (at your option)         licence, soit (à votre gré)
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.cavern.nodes;
+
+import ca.nrc.cadc.util.StringUtil;
+import org.apache.log4j.Logger;
+import org.opencadc.util.fs.ExtendedFileAttributes;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+
+/**
+ * QuotaPlugin implementation that can get and set specific values for interacting with CephFS.  This class relies
+ * on the Extended Attributes to be set on the appropriate folder.
+ */
+public class CephFSQuotaPlugin implements QuotaPlugin {
+    private static final Logger LOGGER = Logger.getLogger(CephFSQuotaPlugin.class);
+    private static final String NAMESPACE = "ceph";
+    private static final String QUOTA_ATTR_KEY = "quota.max_bytes";
+    private static final String RECURSIVE_SIZE_ATTR_KEY = "dir.rbytes";
+
+
+    @Override
+    public Long getBytesUsed(Path directory) {
+        LOGGER.debug("getBytesUsed: " + directory);
+        if (directory == null) {
+            throw new IllegalArgumentException("Cannot get size of null directory");
+        }
+
+        try {
+            final String recursiveBytesUsedValue =
+                    ExtendedFileAttributes.getFileAttribute(directory, CephFSQuotaPlugin.RECURSIVE_SIZE_ATTR_KEY,
+                                                            CephFSQuotaPlugin.NAMESPACE);
+            final long recursiveBytesUsed = Long.parseLong(recursiveBytesUsedValue);
```

Comment:
this is in principle susceptible to NullPointerException (if attr is null for some reason) and NumberFormatException (unlikely).. both runtime so would cause failure.


I think you are right that "warn" and return null is an acceptable behaviour here, so I'd just add those to the catch.